src/components/equations/ItemEffects.ts

```diff
@@ -0,0 +1,29 @@
+import { IMove } from "battlemovr";
+import { GeneralComponent } from "gamestartr";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { IPokemon } from "../Battles";
+
+export class ItemEffects<TGameStartr extends FullScreenPokemon> extends GeneralComponent<TGameStartr> {
+
+    /**
+     * Restores a certain amount of PP to a Pokemon's move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param index   The index of the moveset that is being modified.
+     * @param amount   The amount of PP that is being restored.
+     */
+    public addPP(pokemon: IPokemon, index: number, amount: number) {
```

Comment:
Since whatever calls this is going to have to look up the move to verify anyway, let's just make it take an `IMove` instead of the pokemon and index.